demo-projects/blog/app/lib/apollo.js

```diff
@@ -0,0 +1,62 @@
+// Derived from https://github.com/zeit/next.js/tree/2789e7e0c25c72fbf6be3ef070e530186af14430/examples/with-apollo
+import React from 'react';
+import { ApolloProvider } from '@apollo/react-hooks';
+import createApolloClient from '../apolloClient';
+
+// On the client, we store the Apollo Client in the following variable.
+// This prevents the client from reinitializing between page transitions.
+let globalApolloClient = null;
+
+/**
+ * Always creates a new apollo client on the server
+ * Creates or reuses apollo client in the browser.
+ * @param  {NormalizedCacheObject} initialState
+ * @param  {NextPageContext} ctx
+ */
+const initApolloClient = (initialState, ctx) => {
+  // Make sure to create a new client for every server-side request so that data
+  // isn't shared between connections (which would be bad)
+  if (typeof window === 'undefined') {
+    return createApolloClient(initialState, ctx);
+  }
+
+  // Reuse client on the client-side
+  if (!globalApolloClient) {
+    globalApolloClient = createApolloClient(initialState, ctx);
+  }
+
+  return globalApolloClient;
+};
+
+/**
+ * Creates a withApollo HOC
+ * that provides the apolloContext
+ * to a next.js Page or AppTree.
+ * @returns {(PageComponent: ReactNode) => ReactNode}
+ */
+export const withApollo = () => PageComponent => {
```

Comment:
[1/2] This may be straight from the next.js / apollo examples, but in the context of our examples what's the point of the additional function here? there are no arguments, and it kind of ends up as syntax soup. imo it'll make the example easier to follow without the unnecessary function.

Reply:
Oh yeah, good point! This _used_ to have config options, but I removed them all to simplify it for this demo. Didn't think to remove the extra function. Will remove it 👍